h5grove/utils.py

```diff
@@ -194,27 +194,21 @@ def get_type_metadata(type_id: h5py.h5t.TypeID) -> TypeMetadata:
 
 
 def _sanitize_dtype(dtype: np.dtype) -> np.dtype:
-    """Convert dtype to a dtype supported by js-numpy-parser.
+    """Sanitize numpy dtype to one with a matching typed array in modern JavaScript.
 
-    See https://github.com/ludwigschubert/js-numpy-parser
-
-    :raises ValueError: For unsupported array dtype
+    :raises ValueError: If trying to sanitize a non-numeric numpy dtype
     """
     if dtype.kind not in ("f", "i", "u"):
-        raise ValueError("Unsupported array type")
+        raise ValueError("Expected to sanitize numeric numpy dtype")
```

Comment:
What bothered you in the existing error message ?

Reply:
It didn't specify "numeric", which is what is being checked.

Although it's more opinionated, I also thought it could use a bit more context (i.e. "sanitize") and precision (i.e. with the previous message, one might have thought that the function was encountering an _array_ with an unsupported type, rather than an unsupported numpy dtype object).

Reply:
I see.

I would propose then: `Unsupported numpy dtype encountered when sanitizing {dtype.kind}. Expected a numeric dtype.`

Reply:
Now that I think of it, the context is given by the error stack... :facepalm: I've shortened it down a little and went with printing the full `{dtype}`.

Reply:
> the context is given by the error stack...

When looking at the backend, for sure. But perhaps your improvement is sensible if the only error message is passed to the frontend.

Reply:
I think it'd cause a 500 Server Error, so it would only appear in the logs, wouldn't it?

Reply:
Yeah, I think the mechanism to catch errors and propagate them only works for link resolution errors. So it is fine :+1: